src/content/docs/en/reference/container-reference.mdx

```diff
@@ -165,6 +165,75 @@ const result = await container.renderToString(Card, {
 });
 ```
 
+### `props` option
+
+**Type**: `Record<string, unknown>`
+
+An option to pass [`properties`](en/basics/astro-components/#props) for astro components.
+
+```js name="Card.test.js"
+import Card from "../src/components/Card.astro";
+
+const result = await container.renderToString(Card, { 
+  props: { name: "Hello, world!" },
+});
+```
+
+```astro name="Card.astro"
+---
+// For TypeScript support
+interface Props {
+  name: string;
+};
+
+const { name } = Astro.props;
+---
+<div>
+  {name}
+</div>
+```
+
+#### Type-hints
+
+When using the Container API outside of `.astro` files (e.g. `test.ts`), define 
+and import property interfaces to make them available for type-hints 
+or auto-completion:
+
+```js name="Card.test.ts"
+import Card from "../src/components/Card.astro";
+import type { CardProps } from "../src/components/types.ts";
+
+const props: CardProps = { name: "Hello, world!" };
+const result = await container.renderToString(Card, { 
+  props,
+});
+```
+
+```astro name="types.ts"
+export interface CardProps {
+  name: string;
+};
+```
+
+```astro name="Card.astro"
+---
+import type { CardProps } from "./types.ts";
+
+interface Props extends CardProps {};
+
+const { name } = Astro.props;
+---
+<div>
+  {name}
+</div>
+```
+
+With the dedicated interface for the component properties, astro components 
+can be integrated with any UI frameworks and gain full type-hints support.
+
+Therefore, UI components authors for Astro components can provide a way to
+integrate their components with any UI frameworks with fairly minimal effort!
+
```

Comment:
Sorry if this suggestion seems a bit drastic — DELETE ALL THE THINGS 😅 

I think while it can seem helpful, this is mostly **re**documenting common practices from elsewhere, i.e. how Astro props typing works is not container API specific, and nor is using an imported type for the `props` object. It might be different if we’d made the API take generic parameters (like `renderToString<Props>()` or something), so that we needed to document the specific pattern, but given this is all regular usage, I’d suggest keeping the docs slim and focused is actually more helpful:

```suggestion
```

Reply:
Sure thing. If you want to omit available information on integrating with typescript, that is fine; who needs types or a healthy ecosystem anyway?
The generic type solution is a workaround solution, just like casting the props; I would not spend effort on it when your own JSX types should infer the props correctly - without a dedicated LSP - and thus allow any usage of your Astro components to be correct.

However, if removing the context is a problem (181-195), then all other occurrences on this page should change to keep it consistent. I would suggest not doing that in PR but raising a new one if you desire to change the pattern of the documentation.
Admittedly I have overstepped on the same.

Reply:
I removed the type-hint portion for now and am waiting for your response on changing the reference pattern.